### PR TITLE
ALS recommend all, GLM normal equation, pyspark elastic-net

### PR DESCRIPTION
**Pull request by jkbradley**
Updated ML spark-perf config to have separate test for normal equation solver for GLMs.

Added ALS test for recommendUsersForProducts, recommendProductsForUsers

Renamed config "lbfgs" to "l-bfgs" to match spark.ml Param value.

Added elastic net to PySpark tests.

CC: @mengxr

### CODE REVIEW

File: mllib-tests/v1p5/src/main/scala/mllib/perf/MLAlgorithmTests.scala [hunk 1]
```diff
@@ -237,11 +237,17 @@ class GLMClassificationTest(sc: SparkContext) extends GLMTests(sc) {
     }
 
     if (regType == "elastic-net") {  // use spark.ml
+      val solver = optimizer match {
+        case "l-bfgs" =>
+
```
Comment (mengxr) [inline mllib-tests/v1p5/src/main/scala/mllib/perf/MLAlgorithmTests.scala]: `solver` is not used. It'd be better to say `assert(optimizer == "l-bfgs", "...")`.

File: config/config.py.template [hunk 2]
```diff
@@ -475,34 +447,66 @@ MLLIB_GLM_CLASSIFICATION_TEST_OPTS = MLLIB_CLASSIFICATION_TEST_OPTS + [
...
+        OptionSet("optimizer", ["auto"]),
         # Using elastic-net regularization.
         OptionSet("reg-type", ["elastic-net"]),
         # Runs with L2 (param = 0.0), L1 (param = 1.0).
         OptionSet("elastic-net-param", [0.0, 1.0]),
         # Regularization param (lambda)
         OptionSet("reg-param", [0.01]),
-        # The scale factor for the noise
+        # The scale factor for the noise in feature values
+        OptionSet("scale-factor", [1.0]),
```
Comment (mengxr) [inline config/config.py.template]: Is it easy to change the option name to `noise-level` (or similar)? `scale-factor` is a little confusing in spark-perf.

File: mllib-tests/v1p5/src/main/scala/mllib/perf/MLAlgorithmTests.scala (hunk 1 above)
Comment (jkbradley) [inline mllib-tests/v1p5/src/main/scala/mllib/perf/MLAlgorithmTests.scala]: Oops, thanks

File: config/config.py.template (hunk 2 above)
Comment (jkbradley) [inline config/config.py.template]: OK, renaming:
* scaleFactor, scale -> featureNoise
* epsilon, eps -> labelNoise

**Comment by jkbradley** [comment]
OK updated

**Comment by mengxr** [comment]
Merged into master. Thanks!

**Pull request closed without merging**